lib/File/Compare.pm

```diff
@@ -1,18 +1,14 @@
-package File::Compare;
+package File::Compare 1.1007;
 
-use 5.006;
```

Comment:
In the case it should be `v5.12` and above the package line so older perls don't die on that syntax first?

Reply:
> In the case it should be `v5.12` and above the package line so older perls don't die on that syntax first?

Older perls will never die on that syntax because any older perl will only see the version of File::Compare that ships with that perl.  If we apply this p.r. it will be seen by perl-5.36 but not by any earlier version.

For that reason I feel that we do *not* need any `use VERSION;` statement at the top of the file.  If we want to indicate that the file's syntax is expected to implicitly require, day, perl-5.12 syntax, we can document that in a comment or a commit message.

In order to move the process forward, I would like to apply @JRaspass's revisions in about 3 days.

Reply:
@jkeenan I've restored and bumped the perl version to `v5.12` and left it where it was. This serves two purposes, documenting what level of Perl features this file currently uses, and allows us to drop the `use strict` as we get it from the use version. One day we might even be able to drop the `use warnings` by just bumping said version, that would be nice. Anything that reduces boilerplate for modern perl is very welcome in my book.

Reply:
Agreed, if the purpose is modernization, appropriate use of `use VERSION`  is a good idea and serves several purposes.

Reply:
@JRaspass, you wrote:
> One day we might even be able to drop the use warnings by just bumping said version, that would be nice. A

I just wanted to point out that the one day has come! Since 5.35 this is happening now:

```
$ perl -Mv5.34 -E 'say $a'
$ perl -Mv5.35 -E 'say $a'
Use of uninitialized value $a in say at -e line 1.
```